x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/model/rule_assets/prebuilt_rule_asset.ts

```diff
@@ -10,9 +10,64 @@ import {
   RuleSignatureId,
   RuleVersion,
   BaseCreateProps,
-  TypeSpecificCreateProps,
+  EqlRuleCreateFields,
+  EsqlRuleCreateFields,
+  MachineLearningRuleCreateFields,
+  NewTermsRuleCreateFields,
+  QueryRuleCreateFields,
+  SavedQueryRuleCreateFields,
+  ThreatMatchRuleCreateFields,
+  ThresholdRuleCreateFields,
 } from '../../../../../../common/api/detection_engine/model/rule_schema';
 
+/**
+ * The PrebuiltRuleAsset schema is created based on the rule schema defined in our OpenAPI specs.
+ * However, we don't need all the rule schema fields to be present in the PrebuiltRuleAsset.
+ * We omit some of them because they are not present in https://github.com/elastic/detection-rules.
+ * Context: https://github.com/elastic/kibana/issues/180393
+ */
+const BASE_PROPS_REMOVED_FROM_PREBUILT_RULE_ASSET = zodMaskFor<BaseCreateProps>()([
+  'actions',
+  'throttle',
+  'meta',
+  'output_index',
+  'namespace',
+  'alias_purpose',
+  'alias_target_id',
+  'outcome',
+]);
+
+// `response_actions` is only part of the optional fields in QueryRuleCreateFields and SavedQueryRuleCreateFields
+const TYPE_SPECIFIC_PROPS_REMOVED_FROM_PREBUILT_RULE_ASSET = zodMaskFor<
+  QueryRuleCreateFields | SavedQueryRuleCreateFields
+>()(['response_actions']);
+
+const QueryRuleAssetFields = QueryRuleCreateFields.omit(
+  TYPE_SPECIFIC_PROPS_REMOVED_FROM_PREBUILT_RULE_ASSET
+);
+const SavedQueryRuleAssetFields = SavedQueryRuleCreateFields.omit(
+  TYPE_SPECIFIC_PROPS_REMOVED_FROM_PREBUILT_RULE_ASSET
+);
+
+export const RuleAssetTypeSpecificCreateProps = z.discriminatedUnion('type', [
+  EqlRuleCreateFields,
+  QueryRuleAssetFields,
+  SavedQueryRuleAssetFields,
+  ThresholdRuleCreateFields,
+  ThreatMatchRuleCreateFields,
+  MachineLearningRuleCreateFields,
+  NewTermsRuleCreateFields,
+  EsqlRuleCreateFields,
+]);
```

Comment:
The added complexity worries me a bit, + this is yet another place where we should add a new rule type once it is added to `RuleResponse`. Let's chat over zoom.